relayer/src/connection.rs

```diff
@@ -33,16 +32,45 @@ pub use error::ConnectionError;
 /// Maximum value allowed for packet delay on any new connection that the relayer establishes.
 pub const MAX_PACKET_DELAY: Duration = Duration::from_secs(120);
 
-pub fn from_retry_error(e: retry::Error<ConnectionError>, description: String) -> ConnectionError {
-    match e {
-        retry::Error::Operation {
-            error,
-            total_delay,
-            tries,
-        } => {
-            ConnectionError::max_retry(description, tries, total_delay, Box::new(error.0))
+// Includes utility methods and constants to configure the retry behavior
+// for the channel handshake algorithm.
+mod handshake_retry {
+    use crate::connection::ConnectionError;
+    use crate::util::retry::{clamp_total, ConstantGrowth};
+    use core::time::Duration;
+
+    // Approximate number of retries per block
+    const PER_BLOCK_RETRIES: u32 = 10;
+    // Keep the retry delay constant
+    const DELAY_INCREMENT: u64 = 0;
+    // Maximum retry delay expressed in number of blocks
+    const BLOCK_NUMBER_DELAY: u32 = 10;
+
+    // The default retry strategy.
+    // We retry with a constant backoff strategy. The strategy is parametrized by the
+    // maximum block time expressed as a `Duration`.
+    pub fn default_strategy(max_block_times: Duration) -> impl Iterator<Item = Duration> {
+        let retry_delay = max_block_times / PER_BLOCK_RETRIES;
+
+        clamp_total(
+            ConstantGrowth::new(retry_delay, Duration::from_secs(DELAY_INCREMENT)),
+            retry_delay,
+            max_block_times * BLOCK_NUMBER_DELAY,
+        )
+    }
+
+    pub fn from_retry_error(
```

Comment:
Could we also document this function? I'm not clear on what it's doing.

Reply:
5aba4aa6

Reply:
Does the retry mechanism throw an error every time it retries? Or perhaps only when the maximum number of retries is reached?